### PR TITLE
Merge Andre to Dev

### PR DESCRIPTION
**Pull request by andrbtn**

### CODE REVIEW

File: src/client/grid-view.js [hunk 1]
```diff
@@ -1,539 +1,408 @@
...
-addRow()
-addRow()
-generatePlayingOrder(getNbPlayers())
-//TODO GESTION DE LA MANCHE 10 en front
-
 /**
- * Retourn le type de cellule en fonction de CellType
- * @param cellType
- * @returns {string}
+ * Crée le scordboard à partir du json
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: Please comments in english

File: src/client/grid-view.js [hunk 2]
```diff
@@ -1,539 +1,408 @@
...
+ * @param json
  */
+function createGridView(json) {
+    let grid = new GridView(json);
+    grid.createHeader();
+    grid.createScoreboard();
+    grid.fillScoreboard();
+    console.log(grid.NB_KEELS);
+    grid.generatePossibilities(grid.playerNameToPlay, grid.frameToPlay, grid.throwToPlay);
+    console.log("oui");
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: Is it a test ? if so, please remove it

File: src/client/grid-view.js [hunk 3]
```diff
@@ -1,539 +1,408 @@
...
+    players = null;
+
+    //Data needed to send to API
+    playerNameToPlay = "";
+    frameToPlay = 0;
+    throwToPlay = 0;
+
+    constructor(jsonDocument) {
+        this.gridview = JSON.parse(jsonDocument);
+        console.log( this.gridview)
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: to remove as it's a test

File: src/client/grid-view.js [hunk 4]
```diff
@@ -1,539 +1,408 @@
...
+        // console.log(typeof this.players);
+        // console.log(Object.keys(this.players).length);
+        this.NB_FRAMES = this.gridview['nbFrame'];
+        this.NB_PLAYERS = this.players.length;
+        // this.NB_PLAYERS = this.gridview['players'].length;
+        // this.NB_FRAMES = this.gridview['players'][0]['frames'].length;
+    }
+
+    createHeader() {
+        console.log("Generating header");
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: same, to remove or introduce a bool variable DEBUG

File: src/client/grid-view.js [hunk 5]
```diff
@@ -1,539 +1,408 @@
...
+        let header = document.getElementById("grid-header");
+        let nameCell = document.createElement("th");
+        nameCell.setAttribute("id", this.   getId(0, 0, CellType.NONE));
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: weird space

File: src/client/grid-view.js [hunk 6]
```diff
@@ -1,539 +1,408 @@
...
+        }
+        let total = document.createElement("th");
+        total.textContent = "Total";
+        total.setAttribute("id", this.getId(0, this.NB_FRAMES+1));
+        header.appendChild(total);
+        return header;
+    }
+
+    /**
+     * Retourne l'id d'une cellule (sans vérifier si elle existe)
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: Comments in english please

File: src/client/grid-view.js [hunk 7]
```diff
@@ -1,539 +1,408 @@
...
 
-    //TODO TEMPORARY FIX
-    grid.addPlayer("Player" + rowNumber);
+        const row = "rowP";
+        const col = "_col";
 
-    rowNumber++;
-    return row;
-}
+        const identifierCol = idCol == 0 ? "name" : idCol == this.NB_FRAMES+1 ? "total" : idCol;
```
Comment (pioupiou-devs) [inline src/client/grid-view.js:123]: Can you indent it on multilines and add a comment to explain

File: src/client/grid-view.js [hunk 8]
```diff
@@ -1,539 +1,408 @@
...
-    if (idRow < 0 || idCol < 0 || idCol > 11) {
-        throw new Error(`Invalid id ${idRow}, ${idCol}`);
-    }
-    if (idCol != 10 && cellType == CellType.CENTER) {
-        throw new Error("No center cell for a frame different than 10");
+        if (idRow == 0) {
+            const rowHeader = "row";
+            return rowHeader + idRow + col + identifierCol;
+        } else {
+            const type = cellType != null && idCol != 0 && idCol != 11 ? getCellType(cellType) : "";
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: add a comment to explain

File: src/client/grid-view.js [hunk 9]
```diff
@@ -1,539 +1,408 @@
...
-        firstThrowCell.textContent = first.toString();
-        secondThrowCell.textContent = second.toString();
-    }
+            let total = document.createElement("td");
+            total.setAttribute("id", this.getId(rowNumber, this.NB_FRAMES+1));
+            row.appendChild(total);
+            let tbody = document.querySelector("tbody");
+            tbody.appendChild(row);
+        })
+        console.log({playerNameToPlay: this.playerNameToPlay, frameToPlay: this.frameToPlay, throwToPlay: this.throwToPlay})
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: remove it

File: src/client/grid-view.js [hunk 10]
```diff
@@ -1,539 +1,408 @@
...
+    }
+
+    /**
+     * A partir des données du joueur reçu depuis l'API, remplit le tableau de score
+     */
+    fillScoreboard() {
+        console.log(this.players)
```
Comment (pioupiou-devs) [inline src/client/grid-view.js:209]: remove it

File: src/client/grid-view.js [hunk 11]
```diff
@@ -1,539 +1,408 @@
...
+        this.players.forEach(([player,playerData], index) => {
+            let rowNumber = index+1;
+            console.log(playerData)
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: remove it

File: src/client/grid-view.js [hunk 12]
```diff
@@ -1,539 +1,408 @@
...
-    for (let i = 1; i < rowNumber; i++) {
-        for (let j = 1; j <= 10; j++) {
-            let left = document.getElementById(getId(i, j, CellType.LEFT)).textContent;
-            let right = document.getElementById(getId(i, j, CellType.RIGHT)).textContent;
-            let bottom = document.getElementById(getId(i, j, CellType.BOTTOM)).textContent;
-            let center = document.getElementById(getId(i, j, CellType.CENTER)).textContent;
-            grid["Player"+i](i, j, left, right, bottom, center);
+            button.addEventListener("click", function() {
+                console.log("Click on :", i);
+                //request aussi
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: Why haven't you add the request to the related API ?

File: src/client/grid-view.js [hunk 13]
```diff
@@ -1,539 +1,408 @@
...
-    if (input === "first") {
-        if (!isValidInput("first")) {
-            button.disabled = true;
-            document.getElementById('first').value = "";
-            if (document.getElementById('second').value == "10") {
-                document.getElementById('second').disabled = true;
-                errorMsg.innerHTML = "";
-                block2 = true;
-            }
+let jsonOld = `{
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: to remove if it's a test

File: src/client/grid-view.js [hunk 14]
```diff
@@ -1,539 +1,408 @@
...
-                button.disabled = true;
-                document.getElementById('second').value = "";
-                errorMsg.innerHTML = "Veuillez saisir une valeur pour le premier lancer";
-            } else {
-                button.disabled = false;
-            }
+    ]
+}`;
+
+let jsonNew = `
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: to remove if it's a test

File: src/client/grid-view.js [hunk 15]
```diff
@@ -1,539 +1,408 @@
...
-// };
-
-// module.exports = {
-//     generateHeader, addRow, getId, getCellType, CellType,
-//     getNbPlayers, addScore, nextTurn, generatePlayingOrder, createFrameCell,
-//     checkInput, isValidInput, playingOrder, cellToBePlayed, resetGlobalVariables, rowNumber
-// };
+`
+//get data from backend to initialize
+console.log("jsonNew")
```
Comment (pioupiou-devs) [inline src/client/grid-view.js]: to remove if it's a test

**Review by pioupiou-devs — CHANGES_REQUESTED** [review]

File: src/client/grid-view.js (hunk 12 above)
Comment (andrbtn) [reply src/client/grid-view.js]: couldn't test at the moment

File: src/client/grid-view.js [hunk 16]
```diff
@@ -80,15 +91,14 @@ class GridView {
...
-        return header;
     }
 
     /**
-     * Retourne l'id d'une cellule (sans vérifier si elle existe)
-     * @param idRow ligne/joueur
-     * @param idCol manche
-     * @param cellType type de cellule
-     * @returns {string} id de la cellule
+     * Return the id from a cell given the following argument (sans vérifier si elle existe)
```
Comment (pioupiou-devs) [inline src/client/grid-view.js:106]: Still some french

**Review by pioupiou-devs — CHANGES_REQUESTED** [review]

**Pull request merged**